.github/workflows/cicd.yaml

```
@@ -0,0 +1,98 @@
+name: CI
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{github.repository}}
+
+jobs:
+  lint-dashboard:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup nodejs
+        uses: actions/setup-node@v3
+
+      - name: Install dependencies
+        working-directory: ./dashboard
+        run: npm install
+
+      - run: |
+          cd dashboard &&
+          npm run lint
+
+  lint-worker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+          
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          cache: 'pip'
+        
+      - name: Run flake8
+        uses: py-actions/flake8@v2
+        with:
+          path: "worker"
+  
+  build:
+    name: Build docker image
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - context: dashboard
+            image: dashboard
+          - context: worker
+            model: large-v3
+            image:  worker-large
+          - context: worker
+            model: medium
+            image: worker-medium
+          - context: worker
+            model: small
+            image: worker-small
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{env.IMAGE_NAME}}/${{ matrix.context }}
+          tags: |
+            type=ref, event=pr
+            type=raw, value=${{ matrix.model }}, enable=${{ matrix.image != 'dashboard' }} # tags for model-sizes
+            type=raw, value=latest , enable=${{ github.ref == format('refs/heads/{0}', 'main') && matrix.image == 'worker-large' || matrix.image == 'dashboard'}} # tag worker large and dashboard branch as latest
```

Are you sure that the `||` operator is stronger than the `&&` one? I don't think so.
```suggestion
            type=raw, value=latest , enable=${{ (github.ref == format('refs/heads/{0}', 'main')) && (matrix.image == 'worker-large' || matrix.image == 'dashboard')}} # tag worker large and dashboard branch as latest
```

---

> Are you sure that the `||` operator is stronger than the `&&` one? I don't think so.

Obv. not.

Strictly speaking we wouldn't need the brackets because we are only running the build step on `main`. 

Thanks for the hint, for cleanliness I will add it.